contracts/strategy/yearn/YearnStrategy.sol

```diff
@@ -147,31 +162,52 @@ contract YearnStrategy is IStrategy, AccessControl, Ownable, CustomErrors {
     }
 
     /// @inheritdoc IStrategy
-    function withdrawToVault(uint256 amount)
+    function withdrawToVault(uint256 _amount)
         external
         virtual
         override(IStrategy)
         onlyManager
     {
-        if (amount == 0) revert StrategyAmountZero();
+        if (_amount == 0) revert StrategyAmountZero();
         uint256 uninvestedUnderlying = _getUnderlyingBalance();
 
-        if (amount > uninvestedUnderlying) {
-            uint256 _sharesToWithdraw = _underlyingToShares(
-                amount - uninvestedUnderlying
+        if (_amount > uninvestedUnderlying) {
+            uint256 sharesToWithdraw = _underlyingToShares(
+                _amount - uninvestedUnderlying
             );
 
-            if (_sharesToWithdraw > _getShares())
+            if (sharesToWithdraw > _getShares())
                 revert StrategyNotEnoughShares();
 
             // burn shares and withdraw required underlying to strategy
-            yVault.withdraw(_sharesToWithdraw, address(this), 1);
+            uint256 withdrawnFromYearn = yVault.withdraw(
+                sharesToWithdraw,
+                address(this),
+                maxLossOnWithdraw
+            );
+
+            _amount = uninvestedUnderlying + withdrawnFromYearn;
         }
 
         // transfer underlying to vault
-        underlying.safeTransfer(vault, amount);
+        underlying.safeTransfer(vault, _amount);
+
+        emit StrategyWithdrawn(_amount);
+    }
+
+    /**
+     * Sets the max loss percentage used when withdrawing from the Yearn vault.
+     *
+     * @notice Can only be called by the owner.
+     *
+     * @param _maxLoss The max loss percentage to use when withdrawing from the Yearn vault. Value of 1 equals 0.01% loss.
+     */
+    function setMaxLossOnWithdraw(uint128 _maxLoss) external onlyOwner {
```

Comment:
@gabrielpoca Just thinking about this `onlyOwner` modifier. Should we add a `SETTINGS_ROLE` like in the vault contract and use the `onlyRole` modifier for this function, instead of `onlyOwner`?

Reply:
I have added these [changes](https://github.com/lindy-labs/sc_solidity-contracts/pull/154/commits/1d2573d1116e143365fc7e8fba29168b4a41b954), we can revert the commit if you find them unnecessary